Code/GraphMol/Descriptors/BCUT.cpp

```diff
@@ -101,10 +103,13 @@ std::pair<double, double> BCUT2D(const ROMol &m,
 }
 
 std::vector<double> BCUT2D(const ROMol &m) {
+  unsigned int num_atoms = m.getNumAtoms();
+  PRECONDITION(num_atoms > 0,
```

Comment:
I think these should be ValueErrors, not  preconditions.
It's caused by the user providing bad input, it's not a catastrophic programming error.

Reply:
I switched it to return a vector of zeros, seems the right thing to do and now no one has to worry about try...catch

Reply:
I'm ok with this, but callers still need to do a try...catch since this function with throw if any of the atoms don't have a gasteiger charge property